listenbrainz/webserver/static/js/src/fresh-releases/ReleaseCard.tsx

```diff
@@ -0,0 +1,105 @@
+import * as React from "react";
+import { getAlbumArtFromReleaseMBID } from "../utils/utils";
+import { formattedReleaseDate } from "./utils";
+
+type ReleaseCardProps = {
+  releaseDate: string;
+  artistMBIDs: Array<string>;
+  releaseMBID: string;
+  releaseName: string;
+  artistCreditName: string;
+  releaseTypePrimary: string | undefined | null;
+  releaseTypeSecondary: string | undefined | null;
+  confidence?: number | null;
+};
+
+export default function ReleaseCard(props: ReleaseCardProps) {
+  const {
+    releaseMBID,
+    releaseDate,
+    releaseName,
+    artistMBIDs,
+    artistCreditName,
+    releaseTypePrimary,
+    releaseTypeSecondary,
+    confidence,
+  } = props;
+
+  const COVERART_PLACEHOLDER = "/static/img/cover-art-placeholder.jpg";
+  const RELEASE_TYPE_UNKNOWN = "Unknown";
+
+  const [coverartSrc, setCoverartSrc] = React.useState<string>(
+    COVERART_PLACEHOLDER
+  );
+
+  function releaseTypeTooltip(): string | undefined | null {
+    if (
+      (releaseTypeSecondary !== undefined &&
+        releaseTypePrimary === undefined) ||
+      (releaseTypeSecondary !== null && releaseTypePrimary === null)
+    )
+      return releaseTypeSecondary;
+
+    if (
+      (releaseTypePrimary !== undefined &&
+        releaseTypeSecondary === undefined) ||
+      (releaseTypePrimary !== null && releaseTypeSecondary === null)
+    )
+      return releaseTypePrimary;
+
+    if (
+      (releaseTypePrimary === undefined &&
+        releaseTypeSecondary === undefined) ||
+      (releaseTypePrimary === null && releaseTypeSecondary === null)
+    )
+      return "";
+
+    return `${releaseTypePrimary} + ${releaseTypeSecondary}`;
+  }
+
+  async function getCoverArt() {
+    const coverartURL = await getAlbumArtFromReleaseMBID(releaseMBID);
+    if (coverartURL) {
+      setCoverartSrc(coverartURL);
+    }
+  }
+
+  React.useEffect(() => {
+    getCoverArt();
+  }, []);
```

Comment:
I think this should be the same pattern you found for async calls with useEffect, and passing the required dependencies:

```suggestion
  React.useEffect(() => {
    async function getCoverArt() {
      const coverartURL = await getAlbumArtFromReleaseMBID(releaseMBID);
      if (coverartURL) {
        setCoverartSrc(coverartURL);
      }
    }
    getCoverArt();
  }, [releaseMBID,setCoverartSrc]);
```